_episodes/08-putting-it-all-together.md

```diff
@@ -86,209 +89,235 @@ also change the name of the columns in the DataFrame like this:
+> >                         skiprows=27, delimiter="\t", 
+> >                         names=["agency", "site_id", "datetime",
+> >                                "timezone", "discharge", "discharge_cd"])
+> > discharge["datetime"] = pd.to_datetime(discharge["datetime"])
+> > front_range = discharge[(discharge["datetime"] >= "2013-09-09") & 
+> >                         (discharge["datetime"] < "2013-09-15")]
+> > 
+> > fig, ax = plt.subplots()
+> > front_range.plot(x ="datetime", y="discharge", ax=ax)
+> > ax.set_xlabel("") # no label
+> > ax.set_ylabel("Discharge, cubic feet per second")
+> > ax.set_title(" Front Range flood event 2013")
+> > ~~~
+> > {: .language-python}
+> >
+> > ![Flood event plot](../fig/08_flood_event.png)
+> {: .solution}
+{: .challenge}
 
-    plt.subplot(2,2,2)  # Two row, two columns, position 2
-    plt.plot(t, t**2, 'bs-', label='square')
+### Saving matplotlib figures
 
-    plt.subplot(2,2,3)  # Two row, two columns, position 3
-    plt.plot(t, t**3, 'g^:', label='cubic')
+Once satisfied with the resulting plot, you can save the plot with the `.savefig(*args)` command from matplotlib:
 
-    plt.show()
+~~~
+fig.savefig("my_plot_name.png")    
 ~~~
 {: .language-python}
 
+Which will save the `fig` created using Pandas/matplotlib as a png file with the name `my_plot_name`
+
+> ~~~
+>     Matplotlib recognizes the extension used in the filename and
+>     supports (on most computers) png, pdf, ps, eps and svg formats.
+> ~~~
+{: .callout}
+
+> ## Challenge - Saving figure to file
+> Check the documentation of the `savefig` command and check how 
+> you can comply to journals requiring figures as `pdf` file with
+> dpi >= 300.
+>
+> > ## Answers
+> >
+> > ~~~
+> > fig.savefig("my_plot_name.pdf", dpi=300)
+> > ~~~
+> > {: .language-python}
+> {: .solution}
+{: .challenge}
+
+
 ## Make other types of plots:
 
-Matplotlib can make many other types of plots in much the same way that it makes
-2 dimensional line plots. Look through the examples in
-http://matplotlib.org/users/screenshots.html and try a few of them (click on the
+Matplotlib can make many other types of plots in much the same way that it makes 2 dimensional line plots. Look through the examples in
+<http://matplotlib.org/users/screenshots.html> and try a few of them (click on the
 "Source code" link and copy and paste into a new cell in ipython notebook or
 save as a text file with a `.py` extension and run in the command line).
 
 > ## Challenge - Final Plot
 > Display your data using one or more plot types from the example gallery. Which
 > ones to choose will depend on the content of your own data file. If you are
-> using the streamgage file, you could make a histogram of the number of days with
-> a given mean discharge, use bar plots to display daily discharge statistics, or
-> explore the different ways matplotlib can handle dates and times for figures.
+> using the streamgage file `bouldercreek_09_2013.txt`, you could make a 
```

Comment:
perhaps, add a link to the file `[bouldercreek_09_2013.txt]({{ page.root }}/data/bouldercreek_09_2013.txt)`